pyrtls.pyi

```diff
@@ -0,0 +1,407 @@
+        encryption).
+
+        After this function returns, the connection buffer may not yet be fully flushed.
+        `writable()` can be used to check if the output buffer is empty.
+
+        Mirrors the `RawIO.readinto()` interface.
+        """
+
+    def read_into(self, buf: bytearray) -> int:
+        """
+        Obtain plaintext data received from the peer over this TLS connection.
+
+        If the peer closes the TLS connection cleanly, this returns `0` once all the pending data
+        has been read. No further data can be received on that connection, so the underlying TCP
+        connection should be half-closed too.
+
+        If the peer closes the TLS connection uncleanly (a TCP EOF without sending a
+        `close_notify` alert) this functions raises a `TlsError` exception once any pending data
+        has been read.
+
+        Note that support for `close_notify` varies in peer TLS libraries: many do not support it
+        and uncleanly close the TCP connection (this might be vulnerable to truncation attacks
+        depending on the application protocol). This means applications using pyrtls must both
+        handle EOF from this function, **and** unexpected EOF of the underlying TCP connection.
+
+        If there are no bytes to read, this raises a `TlsError` exception.
+
+        You may learn the number of bytes available at any time by inspecting the return of
+        `process_new_packets()`.
+        """
+
+class ClientConfig:
+    """
+    Create a new `ClientConfig` object (similar to `ssl.SSLContext`). A new `ClientConnection` can
+    only be created by passing in a reference to a `ClientConfig` object.
+
+    The most important configuration for `ClientConfig` is the certificate verification process.
+    Three different options are offered to define the desired process:
+
+    - `platform_verifier` (enabled by default) will enable the platform's certificate verifier
+      on platforms that have on, and searching for CA certificates in the system trust store on
+      other platforms (like Linux and FreeBSD).
+    - `mozilla_roots` will enable a built-in set of Mozilla root certificates. This is independent
+      of the operating system, but depends on the pyrtls package to deliver timely updates.
+    - `custom_roots` allows the caller to specify an iterable of trust anchors. Each item must be:
+      - A `TrustAnchor` object, which is a wrapper around a `webpki::TrustAnchor` object
+      - A `bytes` object containing a DER-encoded certificate
+      - A `str` object containing one PEM-encoded certificate
+
+    The `platform_verifier` option cannot currently be combined with `mozilla_roots` or
+    `custom_roots` (this will raise a `ValueError`), but the latter two can be combined.
+
+    Other options:
+
+    - `alpn_protocols` must be an iterable containing `bytes` or `str` objects, each representing
+      one ALPN protocol string.
+    """
+
+    def __new__(
```

Comment:
`ClientConfig.__new__` and `ServerConfig.__new__` take two arguments that I'm not sure to annotate: 
1. `Iterable[TrustAnchor | str | bytes] | None` or `Iterable[TrustAnchor] | Iterable[str] | Iterable[bytes] | None`?
2. `Iterable[str | bytes]` or `Iterable[str] | Iterable[bytes]`?